src/poll.rs

```diff
@@ -1,3 +1,6 @@
+use sys::time::TimeSpec;
```

Comment:
You have to conditional include this and `SigSet`. This is why the build fails on macos.